content/part4.md

```diff
@@ -0,0 +1,199 @@
+In this blog post, I'll demonstrate how you can build a RESTful API on top of Temporal Workflows, so you can create web apps and mobile apps that store data in Temporal.
+
+## API Setup
+
+For this tutorial, I'll be using [httpx](github.com/bojanz/httpx) along with [mux](https://github.com/gorilla/mux) for routing and [handlers](https://github.com/gorilla/handlers) for CORS.
+
+```go
+package main
+
+import (
+	"context"
+	"github.com/bojanz/httpx"
+	"github.com/gorilla/handlers"
+	"github.com/gorilla/mux"
+	"net/http"
+	"os"
+)
+
+func main() {
+	var err error
+
+  // Set up CORS for frontend
+	var cors = handlers.CORS(handlers.AllowedHeaders([]string{"X-Requested-With", "Content-Type", "Authorization"}), handlers.AllowedMethods([]string{"GET", "POST", "PUT", "HEAD", "OPTIONS"}), handlers.AllowedOrigins([]string{"*"}))
+
+	http.Handle("/", cors(r))
+	server := httpx.NewServer(":"+HTTPPort, http.DefaultServeMux)
+	server.WriteTimeout = time.Second * 240
+
+	err = server.Start()
+	if err != nil {
+		log.Fatal(err)
+	}
+}
+```
+
+The API endpoints will use [Temporal Client methods](https://docs.temporal.io/docs/go/workflows#how-to-start-a-workflow) to create Workflows, and execute Signals and Queries.
+In general, HTTP GET requests execute Queries, HTTP PUT or PATCH requests send Signals, and HTTP POST requests create new Workflows.
+
+```go
+// Create a new cart
+r.Handle("/cart", http.HandlerFunc(CreateCartHandler)).Methods("POST")
+// Get the state of an existing cart
+r.Handle("/cart/{workflowID}/{runID}", http.HandlerFunc(GetCartHandler)).Methods("GET")
+
+// Add a new item to the cart
+r.Handle("/cart/{workflowID}/{runID}/add", http.HandlerFunc(AddToCartHandler)).Methods("PUT")
+// Remove an item from the cart
+r.Handle("/cart/{workflowID}/{runID}/remove", http.HandlerFunc(RemoveFromCartHandler)).Methods("PUT")
+// Update the cart's associated email address
+r.Handle("/cart/{workflowID}/{runID}/email", http.HandlerFunc(UpdateEmailHandler)).Methods("PUT")
+// Check out
+r.Handle("/cart/{workflowID}/{runID}/checkout", http.HandlerFunc(CheckoutHandler)).Methods("PUT")
+```
+
+In this case, the API server and the [Worker](https://docs.temporal.io/docs/go/workers) are separate processes.
+The API server is just an intermediary between the Temporal server and your API server's clients.
+The carts themselves are stored in the Temporal server and in the Worker process.
+
+## Handler Functions
+
+First, let's take a look at the `POST /cart` endpoint. Since we've chosen to represent an individual shopping cart as a Workflow, the `CreateCartHandler()` function will create a new Workflow using `ExecuteWorkflow()`.
+You are responsible for making sure each `POST /cart` call creates a Workflow creates a unique `workflowID`.
+[Temporal treats a Workflow with the same ID as a prior Workflow as a continuation of the prior Workflow](https://docs.temporal.io/docs/shared/continue-as-new).
```

Comment:
I'm not a huge fan of this sentence and phrasing because its a lot more nuanced than just that.

1. It can be configured. This is default behavior.
2. This only works if the last Workflow run completed. Otherwise deduplication comes into play.